typescript/syntax/Statements.sdf3

```diff
@@ -2,19 +2,28 @@ module Statements
 
 imports
 
+  Common
   Declarations
   Expressions
+  Functions
+ 
+context-free priorities
+
+  ReturnStatement.value > ReturnStatement.empty
```

Comment:
It seems that they eagerly try to insert a semi-colon (https://github.com/ternjs/acorn/blob/630508c0948eaf2bd198a9ec0040eb33ec38db2e/src/statement.js#L239-L245) and then try if parsing would work. I think the compromise right now of enforcing a `;` after an empty `return` is valid and easiest for now.